### ISSUE TITLE
AfterControlStatement MultiLine sometimes doesn't work

### ISSUE THREAD

**Issue by realnc** [llvm/llvm-project#51818]
|  |  |
| --- | --- |
| Bugzilla Link | [52476](https://llvm.org/bz52476) |
| Version | 13.0 |
| OS | Linux |

## Extended Description 
With this clang-format file:

```
---
Language: Cpp
Standard: Latest
BasedOnStyle: WebKit
BraceWrapping:
  AfterControlStatement: MultiLine
ColumnLimit: 100
...

```
The following code:

```
void foo()
{
    if (vmType == VM_GGT_TADS2 or vmType == VM_GGT_TADS3 or file_is_savestate(finfo.absoluteFilePath())) {
        bar();
    }
}
```

is formatted incorrectly:

```
void foo()
{
    if (vmType == VM_GGT_TADS2 || vmType == VM_GGT_TADS3
        || file_is_savestate(finfo.absoluteFilePath())) {
        bar();
    }
}

```
What it should be is:

```
void foo()
{
    if (vmType == VM_GGT_TADS2 || vmType == VM_GGT_TADS3
        || file_is_savestate(finfo.absoluteFilePath()))
    {
        bar();
    }
}

```
This isn't the case with all code. It mostly works, but with some code constructs (like the above example) it doesn't.

**Comment by mydeveloperday** [comment]
This may be related to https://github.com/llvm/llvm-project/issues/45978

I think we don't quite calculate the "TotalLength" correctly.. this might be why you say "Sometimes"

**Comment by owenca** [comment]
> With this clang-format file:
>
> ```
> ---
> Language: Cpp
> Standard: Latest
> BasedOnStyle: WebKit
> BraceWrapping:
>   AfterControlStatement: MultiLine
> ColumnLimit: 100
> ...
> ```
You need to add `BreakBeforeBraces: Custom`.

**Comment by owenca** [comment]
Fixed in f4d52cad.
